tests/unit/test_metadata.py

```diff
@@ -0,0 +1,137 @@
+"""
+Unit tests for Metadata class.
+"""
+
+import unittest
+
+import pandas as pd
+from mock import MagicMock
+
+from resdk.resources import Collection, Metadata, Sample
+
+
+class TestMetadata(unittest.TestCase):
+    def setUp(self):
+        self.res = MagicMock()
+
+        # Process
+        self.process = MagicMock(slug="upload-metadata")
+        self.process_unique = MagicMock(slug="upload-metadata-unique")
+
+        # Samples
+        self.sample1 = Sample(resolwe=self.res, id=10)
+        self.sample2 = Sample(resolwe=self.res, id=11)
+
+        # Collection
+        self.collection = Collection(resolwe=self.res, id=5)
+        self.collection._samples = [self.sample1, self.sample2]
+
+        # df
+        self.df = pd.DataFrame(
+            {
+                "Sample ID": [self.sample1.id, self.sample2.id],
+                "Response": ["PD", "CR"],
+            },
+        )
+
+        # Data
+        self.metadata1 = Metadata(
+            resolwe=self.res, id=100, process=self.process, df=self.df
+        )
+        self.metadata2 = Metadata(resolwe=self.res, id=101, process=self.process_unique)
+        self.res.metadata.filter = MagicMock(
+            return_value=[self.metadata1, self.metadata2]
+        )
+
+    def test_init(self):
+        meta = Metadata(self.res, df=self.df)
+
+        self.assertEqual(meta.resolwe, self.res)
+        pd.testing.assert_frame_equal(meta.df, self.df)
+
+    def test_unique(self):
+        # If created from scratch, unique should be True by default
+        meta = Metadata(self.res)
+        # TODO: Why is this false?

```

Comment:
In the `Metadata.__init__` you set `self.unique` to `True` and thereby call `self.unique` setter method which uses the `self.resolwe` which is a `Mock` to set the `self.process` which then also is a `Mock` and in the unique getter you use that Mock for testing which then files. You should rewrite the test or extend the `self.res` `Mock` in the `setUp`

Reply:
Thanks.